### PR TITLE
Add Terraform for AIB without a PIP

### PR DESCRIPTION
**Pull request by ewok-number-3**
We use the Azure Image Builder Service(AIB) to generate the Linux Image. The configuration for the image lives at https://github.com/UKHO/vm-agent-aib

Previously the image has been built by the AIB creating an RG and VM inside the AzDo Live subscription and attaching a PIP to it and running the scripts found in the vm-agent-aib repo. As we cannot create/assign PIPs anymore this process does not work and we cannot add an exclusion to allow a PIP as the RG is generated with a random name each time.

Instead we want to use the new functionality of AIB to use an existing VNet, AIB will setup a LB, proxy VM and private link to be able to create the image without a PIP.

I have set up an example rg (m-az-image-builder-spike) in the AzDo live subscription following the instructions here and the existing VNet instructions above. 

The m-az-image-builder-spike rg could do with being created in Terraform and should be created and destroyed every time the vm-agent-aib pipeline is run.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -32,14 +32,36 @@ pr: none
 
 variables:
 - group: AzDO-variables
-
 jobs:
   - job: GenerateImage
     pool: 'UKHO Ubuntu 1804'
     container: ukhydrographicoffice/terraform-azure:latest
```
Comment (elduddz) [inline azure-pipelines.yml]: pin to 0.12.28?

**Review by elduddz — CHANGES_REQUESTED** [review]
i am aware taht the bit I have commented on isnt necessarily one of your changes.

but it would be worth pinning to a version, as we have found latest will be 0.13 soon and might break a few things.

File: azure-pipelines.yml (hunk 1 above)
Comment (MarcDenman) [inline azure-pipelines.yml]:
```suggestion
    container: ukhydrographicoffice/terraform-azure:0.12.28
```

File: terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,50 @@
+terraform {
+  backend "azurerm" {}
+}
+
+provider "azurerm" {
+  features {}
+}
```
Comment (MarcDenman) [inline terraform/main.tf]:
```suggestion
provider "azurerm" {
  version = "~> 2.17"
  features {}
}
```

**Review by elduddz — APPROVED** [review]

**Pull request merged**